src/cgutils.cpp

```diff
@@ -1106,6 +1106,10 @@ static std::pair<Value*, bool> emit_isa(jl_codectx_t &ctx, const jl_cgval_t &x,
         return std::make_pair(ConstantInt::get(T_int1, *known_isa), true);
     }
 
+    if (is_uniquerep_Type(intersected_type)) {
```

Comment:
This is all wrong now. This says the types are egal, while you used to confirm pointer equivalence.

Reply:
Hmm, ok, so `jl_pointer_egal` used to return true for `Type{Core.TypeofBottom}` whereas this returns false so I thought this is more accurate. I guess that's the wrong way around.... = = .... I guess I'll need to revert to the previous version...

OTOH, what should be the value returned for `Type{Core.TypeofBottom}`? It should match both `Core.TypeofBottom` and `Type{Union{}}` though whenever things hit the compiler it almost always get normalized to `Core.TypeofBottom`

Reply:
Likely neither should return true, since neither is true